tests/test_weaviate_adaptor.py

```diff
@@ -133,42 +150,47 @@ def test_weaviate_adaptor_lifecycle(class_name: str):
         ],
         dtype=np.float32,
     )
-    adaptor.upsert(ids, vectors)
+    adaptor.upsert(to_datapoints(ids, vectors))
 
-    stats = adaptor.stats()  # confirm aggregation reflects inserted objects
+    stats = adaptor.stats()  # GraphQL Aggregate meta-count
     assert stats["ntotal"] == len(ids)
     assert stats["class_name"] == class_name
 
-    query = np.array([1.0, 0.1, 0.0, 0.0], dtype=np.float32)
-    results = adaptor.search(query, topk=3)
+    query = Query(vector=[1.0, 0.1, 0.0, 0.0])
+    results = adaptor.search(query, topk=3)  # GraphQL nearVector search
     assert len(results) == 3
     assert results[0].id == 100
 
-    adaptor.delete([100, 999])
+    adaptor.delete([100, 999])  # data_object.delete
+    expected_total = len(ids) - 1
+    deadline = time.time() + 10.0
     stats_after_delete = adaptor.stats()
-    assert stats_after_delete["ntotal"] == len(ids) - 1
+    while stats_after_delete["ntotal"] != expected_total and time.time() < deadline:
```

Comment:
I don't understand this check. If they are not equal, it should be a failing test since the dataset size is so small. We are also using a dataset of size 4 so we shouldn't need to be waiting at all.

If we're having to wait on Weaviate for such a small dataset, that's either a bug in the adaptor or in Weaviate.